bin/cron/pardot_sync

```diff
@@ -0,0 +1,419 @@
+  # note that user district information seems less good than dashboard.pd_enrollments. So
+  # far no user has had district info where dashboard.pd_enrollments did not. And user district info
+  # has district names like "open csp".
+  districts = District.all.index_by(&:id)
+
+  users = User.where("length(email) > 0")
+
+  users.find_each do |user|
+    PEGASUS_DB[:contacts_rollup].where(:email => user.email).
+        update(:school_name => user.ops_school) unless user.ops_school.nil?
+
+    unless user.district_id.nil?
+      district = districts[user.district_id]
+      unless district.nil?
+        PEGASUS_DB[:contacts_rollup].where(:email => user.email).
+            update(:district_name => district.name)
+      end
+    end
+  end
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+
+  start = Time.now
+  log "Updating district information from dashboard.pd_enrollments"
+  DASHBOARD_REPORTING_DB_READONLY[:pd_enrollments].exclude(:email => nil).exclude(:school_district_id => nil).
+      select_append(:school_districts__name___district_name).select_append(:school_districts__updated_at___district_updated_at).
+      inner_join(:school_districts, :id => :school_district_id).order_by(:district_updated_at).each do |pd_enrollment|
+
+    PEGASUS_DB[:contacts_rollup].where(:email => pd_enrollment[:email]).update(:district_name => pd_enrollment[:district_name], :district_city => pd_enrollment[:city],
+                                                                               :district_state => pd_enrollment[:state], :district_zip => pd_enrollment[:zip])
+  end
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def update_school
+  start = Time.now
+  log "Updating school information from dashboard.pd_enrollments"
+  DASHBOARD_REPORTING_DB_READONLY[:pd_enrollments].exclude(:email => nil).where('length(school) > 0').find do |pd_enrollment|
+    PEGASUS_DB[:contacts_rollup].where(:email => pd_enrollment[:email]).update(:school_name => pd_enrollment[:school])
+  end
+  log "   Completed in " + ((Time.now) - start).to_s + " seconds"
+end
+
+def main
+  return unless only_one_running?(__FILE__)
+
+  start = Time.now
+  PEGASUS_DB.run "SET SQL_SAFE_UPDATES = 0"
+
+  truncate_contacts_rollup
+  insert_from_pegasus_forms
+  insert_from_dashboard_contacts
+  insert_from_dashboard_pd_enrollments
+  insert_from_pegasus_contacts
+  update_roles
+  update_grades_taught
+  update_ages_taught
+  update_district
+  update_school
+  update_facilitator_types
+  update_professional_learning_enrollment
+  update_professional_learning_attendance
+
+  # record contacts' interactions with us based on forms
+  form_infos = [["'CSEdWeekEvent2013'", "hoc_organizer", "'2013'"],
+                ["'HocSignup2014'", "hoc_organizer", "'2014'"],
+                ["'HocSignup2015'", "hoc_organizer", "'2015'"],
+                ["'HocSignup2016'", "hoc_organizer", "'2016'"],
+                ["'Petition'", "roles", "'Petition Signer'"]
```

Comment:
I'd prefer this as a constant, defined at the top of the script. Also add a comment explaining what the values are, or change it to an array of hashes (so it's self explanatory), e.g. `{form: 'Petition', field: 'roles', value: 'Petition Signer'}`